automation/docker/run-tests.sh

```diff
@@ -1,73 +1,79 @@
 #!/bin/bash
 
+# Get script location, via https://stackoverflow.com/a/12197227
+pushd . > /dev/null
+WORKING_DIR="${BASH_SOURCE[0]}";
+  while([ -h "${WORKING_DIR}" ]) do
+    cd "`dirname "${WORKING_DIR}"`"
+    WORKING_DIR="$(readlink "`basename "${WORKING_DIR}"`")";
+  done
+cd "`dirname "${WORKING_DIR}"`" > /dev/null
+WORKING_DIR="`pwd`";
+popd  > /dev/null
+
+if [ -z ${1+x} ]; then
+  printf "Must specify where Firecloud is running: 'fiab', 'local', 'alpha', 'prod', or ip."
+  exit 1
+fi
+
+# Defaults
+ENV=dev
+NUM_NODES=2
+TEST_ENTRYPOINT="testOnly -- -l ProdTest"
+TEST_CONTAINER="automation-$(cat /dev/urandom | env LC_CTYPE=C tr -dc 'a-z0-9' | fold -w 8 | head -n 1)"
+
 # Parameters
-NUM_NODES="${1:-4}"  # default to 4
-ENV="${2:-dev}"  # default to dev
-export ENV=$ENV
-HUB_COMPOSE=hub-compose-fiab.yml
-
-#if test runs against a remote FIAB on a GCE node, put IP in param 3
-#if test runs against a local FIAB on a Docker, put "local" in param 3
-#else leave blank (test runs against a non FIAB host, such as real dev)
-DOCKERHOST="127.0.0.1"
-DOCKERHOST=${3:-$DOCKERHOST}
-if [ $DOCKERHOST = "alpha" -o $DOCKERHOST = "prod" ];
-  then
-    DOCKERHOST=
-    HUB_COMPOSE=hub-compose.yml
+FC_INSTANCE=${1}
+ENV=${2:-$ENV}
+VAULT_TOKEN=${3:-$(cat ~/.vault-token)}
+
+export FC_INSTANCE WORKING_DIR ENV
+
+if [ "$ENV" = "prod" ]; then
+  TEST_ENTRYPOINT="testOnly -- -n ProdTest"
 fi
-export DOCKERHOST=$DOCKERHOST
-TEST_CONTAINER=${4:-automation}
-VAULT_TOKEN=$5
-WORKING_DIR=${6:-$PWD}
-export WORKING_DIR=$WORKING_DIR
-
-if [ -z $VAULT_TOKEN ]; then
-    VAULT_TOKEN=$(cat ~/.vault-token)
+
+if [ "$FC_INSTANCE" = "local" ]; then
+  FC_INSTANCE="172.19.0.1"
 fi
 
-if [ "$DOCKERHOST" = "local" ]
-  then
-    docker pull chickenmaru/nettools
-    export DOCKERHOST=`docker run --net=docker_default -it --rm chickenmaru/nettools sh -c "ip route|grep default|cut -d' ' -f 3"`
-    echo "Docker Host from Docker Perspective: $DOCKERHOST"
+if [ "$FC_INSTANCE" = "fiab" ]; then
+  FC_INSTANCE="$(grep 'firecloud-fiab.dsde-dev.broadinstitute.org' /etc/hosts | awk '{print $1}')"
```

Comment:
sometimes i have multiple entries in my /etc/hosts but comment out the one i'm not using. i'm guessing this won't skip the commented out lines

Reply:
Hmm, yeah. In that case, you'd have to just put in the IP address as the argument.

Reply:
Option:
```
FC_INSTANCE="$(cat /etc/hosts | grep -v '#' | grep 'firecloud-fiab.dsde-dev.broadinstitute.org' | awk '{print $1}')"
```